core/text/src/markdown/lexer.rs

```diff
@@ -0,0 +1,218 @@
+use nom::branch::*;
+use nom::bytes::complete::*;
+use nom::character::complete::*;
+use nom::combinator::*;
+use nom::multi::*;
+use nom::sequence::*;
+use nom::{IResult, InputIter, InputLength, InputTake, Slice};
+use std::iter::Enumerate;
+use std::ops::{Range, RangeFrom, RangeFull, RangeTo};
+use std::slice::Iter;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum LexToken {
+    ControlWordStarter,
+    LBrace,
+    RBrace,
+    Space(String),
+    Word(String),
+}
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct Tokens<'a> {
+    pub tok: &'a [LexToken],
+    start: usize,
+    end: usize,
+}
+
+impl<'a> Tokens<'a> {
+    #[allow(clippy::ptr_arg)]
+    pub fn new(vec: &'a Vec<LexToken>) -> Tokens<'a> {
```

Comment:
This function should be able to take a slice instead of a `Vec`, if I'm not mistaken?

Reply:
Normally yes, however it's an internal function and the only place that it is used is internally where the only sequence of LexTokens that are encountered are `Vec`s from `lex_input`.